src/vellum/workflows/nodes/utils.py

```diff
@@ -2,28 +2,24 @@
 from typing import Type
 
 from vellum.workflows.nodes import BaseNode
-from vellum.workflows.references import NodeReference
 from vellum.workflows.types.generics import NodeType
 
 ADORNMENT_MODULE_NAME = "<adornment>"
 
 
 @cache
 def get_wrapped_node(node: Type[NodeType]) -> Type[BaseNode]:
-    if hasattr(node, "subworkflow"):
-        subworkflow = node.subworkflow
-        if isinstance(subworkflow, NodeReference) and subworkflow.instance:
-            graph = subworkflow.instance.graph
-            if issubclass(graph, BaseNode):
-                return graph
+    wrapped_node = getattr(node, "__wrapped_node__", None)
+    if wrapped_node is None:
+        raise AttributeError("Wrapped node not found")
 
-    raise TypeError("Wrapped subworkflow contains more than one node")
+    return wrapped_node
```

Comment:
in a separate pass, I think we could replace all uses of `get_wrapped_node`/`has_wrapped_node` with just this attribute